### ISSUE TITLE
backup.tar.xz not reflecting what was backed up

### ISSUE THREAD

**Issue by roastlechon** [boombatower/docker-backup#1]
Hello,
I was testing out docker-backup and can't seem to backup the volume container.

Here is an example of what I am running.

```
sudo docker run -d -v /var/lib/mysql --name mysql_data busybox true
// assume something was written to /var/lib/mysql
sudo docker pull boombatower/docker-backup
sudo docker run --rm --volumes-from mysql_data -v $(pwd):/backup boombatower/docker-backup backup
ls -lha
-rw-r--r-- 1 ubuntu ubuntu  300 Mar 17 01:21 backup.tar.xz
```

I would love to use this, but I can't get it working :(

**Comment by boombatower** [comment]
Can you post the contents of backup.tar.xz or explain how "backup.tar.xz not reflecting what was backed up"?

**Comment by roastlechon** [comment]
The contents were pretty much this:

```
ubuntu@ip-172-31-48-82:~/backup$ sudo docker run --rm --volumes-from bamboo_data -v $(pwd):/backup boombatower/docker-backup backup
Volumes detected: /backup
Creating archive...
tar: Removing leading `/' from member names
tar: /backup: file changed as we read it
Written to /backup/backup.tar.xz
ubuntu@ip-172-31-48-82:~/backup$ ls -lha
total 12K
drwxrwxr-x  2 ubuntu ubuntu 4.0K Mar 17 20:16 *.*
drwxr-xr-x 12 ubuntu ubuntu 4.0K Mar 17 00:56 *..*
-rw-r--r--  1 root   root    320 Mar 17 20:16 *backup.tar.xz*
ubuntu@ip-172-31-48-82:~/backup$ tar xfv backup.tar.xz
backup/
backup/backup.tar.xz
ubuntu@ip-172-31-48-82:~/backup$ ls -lha backup/
total 8.0K
drwxrwxr-x 2 ubuntu ubuntu 4.0K Mar 17 01:31 *.*
drwxrwxr-x 3 ubuntu ubuntu 4.0K Mar 17 20:16 *..*
-rw-r--r-- 1 ubuntu ubuntu    0 Mar 17 20:16 *backup.tar.xz*
ubuntu@ip-172-31-48-82:~/backup$
```

**Comment by roastlechon** [comment]
For some reason, markdown wasnt working through email reply...

**Comment by boombatower** [comment]
Looks like it is trying to backup the `/backup` mount itself. Which means the output of `cat /proc/mounts` must differ from what I have been getting. I could change the script to ignore by name instead of how it works now. Could you paste the output of that command:

```
docker run --rm --volumes-from mysql_data -v $(pwd):/backup boombatower/opensuse /bin/bash -c "cat /proc/mounts"
```

I assumed this would be the same given that the script is running in the container I provided, but perhaps it is not and is based on host kernel which makes some sense I suppose.

To clarify, I tried your setup using the following commands and it worked as expected

```
docker run -d -v /var/lib/mysql --name mysql_data busybox true

docker run -it --volumes-from mysql_data opensuse /bin/bash
bash-4.2# cd /var/lib/mysql/
bash-4.2# touch test
bash-4.2# touch test2
bash-4.2# ls
test  test2
bash-4.2# exit
exit

docker run --rm --volumes-from mysql_data -v $(pwd):/backup boombatower/docker-backup backup
Volumes detected: /var/lib/mysql 
Creating archive...
tar: Removing leading `/' from member names
Written to /backup/backup.tar.xz
```

The generated tar container the `/var/lib/mysql` structure with `test` and `test2` inside.

In the output you provided I see `Volumes detected: /backup` which seems to indicate the output is different and thus detect `/backup` mount instead of intended `--volumes-from` mounts. Probably a simple fix either sorting output, or checking paths against blacklist instead of current approach.

**Comment by roastlechon** [comment]
Here is the output:

$ sudo docker run --rm --volumes-from mysql_data -v $(pwd):/backup
boombatower/opensuse /bin/bash -c "cat /proc/mounts"
Unable to find image 'boombatower/opensuse' locally
Pulling repository boombatower/opensuse
aad975497e2f: Download complete
511136ea3c5a: Download complete
a113d52c65c3: Download complete
758c78b4040c: Download complete
434d5a33563e: Download complete
rootfs / rootfs rw 0 0
/dev/mapper/docker-202:1-398196-0f46d48e673d673a430bab3da4e9e0f53b21c28a3c98908aaad07b09fe065667
/ ext4 rw,relatime,discard,stripe=16,data=ordered 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
sysfs /sys sysfs ro,relatime 0 0
tmpfs /dev tmpfs rw,nosuid,mode=755 0 0
shm /dev/shm tmpfs rw,nosuid,nodev,noexec,relatime,size=65536k 0 0
devpts /dev/pts devpts
rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=666 0 0
/dev/disk/by-uuid/e1d70192-1bb0-461d-b89f-b054e45bfa00 /.dockerinit ext4
ro,relatime,discard,data=ordered 0 0
tmpfs /etc/resolv.conf tmpfs ro,relatime,size=404684k,mode=755 0 0
/dev/disk/by-uuid/e1d70192-1bb0-461d-b89f-b054e45bfa00 /etc/hostname ext4
ro,relatime,discard,data=ordered 0 0
/dev/disk/by-uuid/e1d70192-1bb0-461d-b89f-b054e45bfa00 /etc/hosts ext4
ro,relatime,discard,data=ordered 0 0
/dev/disk/by-uuid/e1d70192-1bb0-461d-b89f-b054e45bfa00 /var/lib/mysql ext4
rw,relatime,discard,data=ordered 0 0
/dev/disk/by-uuid/e1d70192-1bb0-461d-b89f-b054e45bfa00 /backup ext4
rw,relatime,discard,data=ordered 0 0
proc /proc/sys proc ro,nosuid,nodev,noexec,relatime 0 0
proc /proc/sysrq-trigger proc ro,nosuid,nodev,noexec,relatime 0 0
proc /proc/irq proc ro,nosuid,nodev,noexec,relatime 0 0
proc /proc/bus proc ro,nosuid,nodev,noexec,relatime 0 0
tmpfs /proc/kcore tmpfs rw,nosuid,mode=755 0 0

**Comment by boombatower** [comment]
Fixed by #2
